server/src/main/java/io/crate/protocols/postgres/PgClient.java

```diff
@@ -475,13 +482,16 @@ public static AuthType of(int type) {
 
         private final SslContext sslContext;
         private final String user;
+        private final SSLMode sslMode;
 
         private boolean expectSSLResponse;
 
-        public Decoder(String user, @Nullable SslContext sslContext) {
+
+        public Decoder(String user, SSLMode sslMode, @Nullable SslContext sslContext) {
             this.user = user;
+            this.sslMode = sslMode;
             this.sslContext = sslContext;
-            this.expectSSLResponse = sslContext != null;
+            this.expectSSLResponse = sslMode == SSLMode.REQUIRE || sslMode == SSLMode.PREFER;
```

Comment:
Can't it be that  `sslCtx == null` because of an Exception? Also, if `sslContext ` is `null` and mode is `REQUIRE` we probaby need to close connection and report smth like "Could't setup secured connection but SSL is required" (phrasing could be better)

Reply:
Right, thanks for catching. I added a fixup that should address it. SSL handshake may fail with the fallback config, but it should result in a (somewhat) meaningful error if it does

Reply:
Minor: If I understood fixup correctly,  SslContext is not `@Nullable` anymore